Framework/Dotzu/Dotzu/ManagerListLogViewController.swift

```diff
@@ -81,7 +82,54 @@ class ManagerListLogViewController: UIViewController {
         LogNotificationApp.resetCountBadge.post(Void())
     }
 
-    @IBAction func resetLogs(_ sender: Any) {
+    @IBAction func showActions() {
+
+        let alertController = UIAlertController(title: "More actions",
+                                                message: nil,
+                                                preferredStyle: .actionSheet)
+
+        let sendLogsButton = UIAlertAction(title: "Email logs",
+                                           style: .default,
+                                           handler: { [weak self] (action) in self?.emailLogs() })
+        alertController.addAction(sendLogsButton)
+
+        let resetLogsButton = UIAlertAction(title: "Reset logs & network",
+                                            style: .destructive,
+                                            handler: { [weak self] (action) in self?.resetLogs() })
+        alertController.addAction(resetLogsButton)
+
+        let cancelButton = UIAlertAction(title: "Cancel", style: .cancel, handler:nil)
+        alertController.addAction(cancelButton)
+
+        self.navigationController?.present(alertController, animated: true, completion: nil)
+
+    }
+
+    private func emailLogs() {
+
+        if MFMailComposeViewController.canSendMail() {
+
+            let mail = MFMailComposeViewController()
+            mail.mailComposeDelegate = self;
+            let appName = Bundle.main.infoDictionary?[kCFBundleNameKey as String] as? String
+            mail.setSubject("\(appName ?? "App") Log")
+
+            var formattedData = Data()
+            for itemIndex in 0...self.dataSourceLogs.count {
+                if let log = self.dataSourceLogs[itemIndex] {
+                    let format = LoggerFormat.format(log: log)
+                    let data = Data(("Log Index:\(itemIndex)\n \(format.str) \n\n").utf8)
+                    formattedData.append(data)
+                }
+            }
+
+            mail.addAttachmentData(formattedData, mimeType: "text/plain", fileName: "log-\(Date())")
+
+            self.present(mail, animated: true, completion: nil)
```

Comment:
Sounds good 👍 But what if you have a huge list of logs?
It's also the problem I am having in that current version.

Reply:
I did load test for 4000 logs. 
* Create 4000 logs at once. App crash ~2000 logs due to out of memory issue.
* Create 400 logs at once for 10 times. Created 4000 logs successfully.
** I did notice CPU usage spike but it went down instantly after logs are finished.

File size for 4000 logs was less than 0.5 MB. Mail app attachment limit is 20MB, so we are good on size :) Also, I did not see any screen freezing or performance issue while creating logs file.

I think, Logging 4000/400 logs at once are not real life scenarios and CPU spike is expected in this case. I will be pretty happy with Logger & email system which handles 4000+ events. That should give you enough data to look for. 

Thoughts?